### ISSUE TITLE
Raspberry Pi4 bullseye

### ISSUE THREAD

**Issue by szukw000** [fltk/fltk#461]
With Raspberry Pi4 (bullseye) I found 2 errors during compilation.

Error 1 could be repaired by adding

    extern char *fl_get_font_xfld(int font, int size);

in line 52 of file 'src/drivers/X11/Fl_X11_Screen_Driver.cxx'.

Error 2 I circumvented with

     -DFLTK_BUILD_EXAMPLES=off -DFLTK_BUILD_TEST=off

With these  changes the compilation succeeded.

winfried
[fltk-compile-error-1.txt.gz](https://github.com/fltk/fltk/files/9070837/fltk-compile-error-1.txt.gz)
[fltk-compile-error-2.txt.gz](https://github.com/fltk/fltk/files/9070839/fltk-compile-error-2.txt.gz)

**Comment by Albrecht-S** [comment]
Thanks for the report. Error 1 might be caused by the way you build FLTK (this is not your fault) and we need more info about your build process (tools, commandline, etc.). The log helps but is not enough, please provide more info so we can reproduce the issue. I assume that it is not raspberry/bullseye related but depends on the build tools and options. Please give us more info (I see that you're using CMake but I need the full commandline).

Error 2 is a known issue that's caused by building FLTK shared libraries which is not yet working successfully. I'm working on this issue - the only workarounds here are either not building shared libraries or - as you did - not building the test programs at all. Anyway, for this error your full build commandline would be helpful as well. Thanks in advance.

**Comment by Albrecht-S** [comment]
PS: just in case, please provide the CMake configure log as well, i.e. the full CMake commandline plus the log before you start the build process. Thanks.

**Comment by szukw000** [comment]
@Albrecht-S ,
I added the log.
winfried
[fltk-1.4.x-2022-07-02-cmake-log_and_error.txt.zip](https://github.com/fltk/fltk/files/9077534/fltk-1.4.x-2022-07-02-cmake-log_and_error.txt.zip)

**Comment by Albrecht-S** [comment]
Thanks for the log, without the log I wouldn't have been able to find the cause. Error 1 was caused by the fact that your build didn't find `Xft` headers and/or the `Xft` libraries. It can be reproduced by disabling `Xft` in the build, for instance by using CMake with `-DOPTION_USE_XFT:BOOL=OFF`.

Building w/o `Xft` is not recommended, you should try to install `Xft` headers. The Debian package name is very likely `libxft-dev` (`apt install libxft-dev`).

This bug is fixed in commit 79ce6547ce78c765290db574f1ff4930de862ba0. Thanks for the report.

**Comment by Albrecht-S** [comment]
Error 2 is caused by a combination of effects:
1. your build system doesn't have the needed OpenGL headers
2. building shared libs with OpenGL does not honor the missing OpenGL libs

Both together cause that unittests-shared is built but misses OpenGL libs. I don't know yet how to fix this - i.e. the build issue if OpenGL is not found or not supported.

OTOH, if you want or need OpenGL support you'd need to install OpenGL headers anyway, so this could help. Your log reports:
```
-- Could NOT find OpenGL (missing: OPENGL_opengl_LIBRARY OPENGL_glx_LIBRARY OPENGL_INCLUDE_DIR) 
```

This is not a (fatal) configure error, the build should succeed anyway w/o using OpenGL. On my system with installed OpenGL files it builds and runs OK. I'll look further into it.

I believe Debian package `libgl1-mesa-dev` would install the needed headers so your build can finish. Hopefully.

But again, the build should finish anyway, so thanks for the report.

**Comment by Albrecht-S** [comment]
@szukw000 I'm pretty sure that this issue has been fixed meanwhile in FLTK 1.4.0 (Git master).

Can you verify that it's fixed? If not, please let us know with details about your build.

Anyway, I'm closing this issue now. You can still add comments and request to reopen if it's **_not_** fixed. Thanks.

**Comment by Albrecht-S** [comment]
This issue should have been closed on Jan 9, 2024, see comment above. Closing now.
